src/main/java/cs2103/v15_1j/jimjim/command/UndoCommand.java

```diff
@@ -0,0 +1,32 @@
+package cs2103.v15_1j.jimjim.command;
+
+import java.util.Stack;
+
+import cs2103.v15_1j.jimjim.model.DataLists;
+import cs2103.v15_1j.jimjim.searcher.Searcher;
+import cs2103.v15_1j.jimjim.storage.Storage;
+
+class UndoCommand implements Command {
+
+	@Override
+	public String undo(DataLists displayList, DataLists masterList, Storage storage, Searcher searcher) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public String execute(DataLists displayList, DataLists masterList, Storage storage, Searcher searcher) { 
+		return null;
+	}
+	
+	public String execute(DataLists displayList, DataLists masterList, Storage storage, 
```

Comment:
this function is not in the Command interface, so it is invisible to the controller. I think you should just change the interface to add this `Stack<Command>` to the parameters of `execute`